experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -108,42 +108,6 @@ export function valueSchemaAllows<TSchema extends ValueSchema>(
 	}
 }
 
-/**
- * Use for readonly view of Json compatible data that can also contain IFluidHandles.
- *
- * Note that this does not robustly forbid non json comparable data via type checking,
- * but instead mostly restricts access to it.
- */
-export type FluidSerializableReadOnly =
-	| IFluidHandle
-	| string
-	| number
-	| boolean
-	// eslint-disable-next-line @rushstack/no-new-null
-	| null
-	| readonly FluidSerializableReadOnly[]
-	| { readonly [P in string]?: FluidSerializableReadOnly };
-
-// TODO: replace test in FluidSerializer.encodeValue with this.
-export function isFluidHandle(value: undefined | FluidSerializableReadOnly): value is IFluidHandle {
-	if (typeof value !== "object" || value === null) {
-		return false;
-	}
-	const handle = (value as Partial<IFluidHandle>).IFluidHandle;
-	// Regular Json compatible data can have fields named "IFluidHandle" (especially if field names come from user data).
-	// Separate this case from actual Fluid handles by checking for a circular reference: Json data can't have this circular reference so it is a safe way to detect IFluidHandles.
-	const isHandle = handle === value;
```

Comment:
So this depends on the implementation of `IProvideFluidHandle` to return itself and not delegate the `.IFluidHandle` object to something else, right?  Having trouble reasoning over if that's risky or not.

Reply:
Wow I'm not red-green colorblind but I definitely felt like I was reading new code here.  Haha still applies, to the green copy of it below...

Reply:
Ok yeah I was concerned about something slightly different, but was confused. This is fine given the current scheme.